src/nemo_evaluator/environments/gym.py

```
@@ -396,56 +408,83 @@ def _build_cmd(self) -> list[str] | str:
-            except (urllib.error.URLError, OSError):
-                pass
-            else:
+            probe_timeout = min(2.0, remaining)
+            for probe in ("/health", "/openapi.json"):
                 try:
-                    with urllib.request.urlopen(
-                        f"{self.endpoint}/openapi.json",
-                        timeout=2.0,
-                    ) as r2:
-                        if r2.status == 200:
+                    with urllib.request.urlopen(f"{self.endpoint}{probe}", timeout=probe_timeout) as r:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)